python3Packages.apache-beam: 2.66.0 -> 2.67.0

GaetanLepage opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

Diff: https://github.com/apache/beam/compare/v2.66.0...v2.67.0
Changelog: https://github.com/apache/beam/blob/release-v2.67.0/CHANGES.md

cc @ndl


- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python3Packages.apache-beam: 2.66.0 -> 2.67.0

Diff: https://github.com/apache/beam/compare/v2.66.0...v2.67.0

Changelog: https://github.com/apache/beam/blob/release-v2.67.0/CHANGES.md

dotlambda commented on pkgs/development/python-modules/apache-beam/default.nix:
```diff
@@ -62,16 +63,31 @@
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail "distlib==0.3.7" "distlib" \
+      --replace-fail "yapf==0.43.0" "yapf" \
+      --replace-fail "grpcio-tools==1.62.1" "grpcio-tools" \
+      --replace-fail "grpcio-tools==1.71.0" "grpcio-tools" \
+      --replace-fail "mypy-protobuf==3.5.0" "mypy-protobuf" \
```
Does
```suggestion
      --replace-fail "==" ">=" \
```
work?

GaetanLepage replied:
Had to handle `numpy`'s `<2.3.0`, but it worked for the rest.

GaetanLepage committed:
python3Packages.mlcroissant: 1.0.21 -> 1.0.22

Diff: https://github.com/mlcommons/croissant/compare/v1.0.21...v1.0.22

Changelog: https://github.com/mlcommons/croissant/releases/tag/v1.0.22
